src/dynamic_graph_head/sim_head.py

```diff
@@ -105,12 +111,19 @@ def read(self):
             # Write to the measurement history with noise.
             history['joint_positions'][write_idx] = q[7:]
             history['joint_velocities'][write_idx] = dq[6:]
-            history['imu_gyroscope'][write_idx] = dq[3:6]
-
+            history['imu_gyroscope'][write_idx] = self._robot.get_base_imu_angvel()
+            history['imu_accelerometer'][write_idx] = self._robot.get_base_imu_linacc() 
             self._sensor_imu_gyroscope[:] = history['imu_gyroscope'][read_idx]
-
+            self._sensor_imu_accelerometer[:] = history['imu_accelerometer'][read_idx]
             self._sensor__vicon_base_position[:] = q[:7]
             self._sensor__vicon_base_velocity[:] = dq[:6]
+            # only read forces for free floating for now 
+            contact_status, contact_forces = self._robot.get_force()
+            for i, cnt_id in enumerate(self._robot.pinocchio_endeff_ids):
+                self._sensor__force_plate_force[i,:] = contact_forces[i][:]
+                self._sensor__force_plate_status[i] = contact_status[i] 
```

Comment:
I am not sure putting more and more fields on the sim-head that are then read/processed by other objects is a good idea. But maybe for now we can keep it.

Reply:
I tried to follow the same way you added the vicon to avoid different logic